src/main/webapp/scripts/perspective.js

```diff
@@ -1,21 +1,26 @@
+loadScript('scripts/classes/comment-class.js');
+loadScript('scripts/perspective-api-service.js');
+
+NO_COMMENT_ERROR =  new Error('No comment inputted');
+COMMENT_LENGTH_EXCEEDED_ERROR = new Error('Comment is too long. We are currently only able to analyze comments of up to approx. 3000 characters.');
+PERSPECTIVE_API_ERROR =  new Error('The analysis feature is unavailable at this moment.');
+
 function getToxicity() {
-  API_KEY = '';
-  const commentText = document.getElementById("comment-text").value;
-  const commentObject = {
-    'comment': {'text': commentText},
-    'requestedAttributes': {'TOXICITY': {}},
-    'languages': ['en']
-  };
-  const json = JSON.stringify(commentObject);
-  const request = new Request(
-    `https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=${API_KEY}`,
-    {method: 'POST', body: json}
-  );
-  fetch(request)
-    .then(response => response.json())
-    .then((responseJson) => {
-      const score = responseJson.attributeScores.TOXICITY.summaryScore.value;
-      const containerDOM = document.getElementById('toxicity-container');
-      containerDOM.innerText = (score*100).toFixed(2) + '% likely to be toxic.';
-    });
+  let comment;
+  try {
+    comment = new Comment(getDomValue('comment-text'));
+  } catch(error) {
+    updateDom(error.message, 'toxicity-container');
+  }
+  perspectiveApiCall(comment, function(error, result) {
+    if(error != null) {
+      updateDom(error.message, 'toxicity-container');
+      return;
+    }
```

Comment:
Handling errors in callback functions is kinda weird to me, but I wanted to stick with the pattern of this file handing calling the DOM functions. I found [this](https://medium.com/@1sherlynn/how-to-handle-errors-in-asynchronous-javascript-code-when-working-with-callbacks-dcd32bca4b6b) article that told me to use this pattern, which is a familiar pattern from NodeJS projects I have worked on. Let me know if there are better preferred patterns though